src/Twig/SourceCodeExtension.php

```diff
@@ -92,8 +93,17 @@ private function getCallableReflector(callable $callable): \ReflectionFunctionAb
         return new \ReflectionFunction($callable);
     }
 
-    private function getTemplateSource(TemplateWrapper $template): array
+    private function getTemplateSource($template): array
     {
+        if (!$template instanceof TemplateWrapper && !$template instanceof Template) {
+            throw new \Exception(sprintf(
```

Comment:
I'm not sure this exception is needed... this function is private/local and `$twig->resolveTemplate($template)` ony returns `TemplateWrapper` or `Template`.

I'd suggest adding a PHPDoc block instead.

Reply:
I'm fine with PHPDoc, just was coding defensively... Let's wait for more feedback then

Reply:
I agree with Yonel on this one.

Reply:
OK, it's simpler now 👍